src/components/Avatar/Avatar.tsx

```diff
@@ -1,38 +1,28 @@
-import styles from "./Avatar.module.css"
+import { AvatarBase, AvatarBaseProps } from "./AvatarBase"
 import { getInitials } from "./utils/getInitials"
 
-export type AvatarSize = "sm" | "md" | "lg"
-
-interface AvatarProps {
+export interface AvatarProps
+  extends Omit<AvatarBaseProps, "children" | "ariaLabel"> {
   name: string
-  size?: AvatarSize
-  src?: string
 }
 
-export function Avatar({ name, size = "sm", src }: AvatarProps) {
+export function Avatar({ name, ...avatarBaseProps }: AvatarProps) {
   const Child = () => {
-    if (src) {
+    if (avatarBaseProps.src) {
       return null
     } else {
       const initials = getInitials(name)
-      return <div className={styles.text}>{initials}</div>
+      return <>{initials}</>
     }
   }
```

Comment:
Naming: Child seems too generic, everything is a child in react. Are there simpler alternatives to get this implemented?
This might not be simpler, but one idea would be to replace Child with an expression:
```tsx
{ hasImage ? null : getInitials(name) }
```
Or if you don't want to use ternaries
```tsx
{ doesNotHaveImage && getInitials(name) }
```